tree-sitter/src/TreeSitter/Unmarshal.hs

```diff
@@ -210,7 +214,7 @@ instance GUnmarshal U1 where
 
 -- For regular leaf nodes
 instance {-# OVERLAPPABLE #-} GUnmarshal (M1 S s (K1 c Text.Text)) where
-  gunmarshalNode = M1 . K1 <$> unmarshalNode
+  gunmarshalNode = M1 . K1 <$> (peekNode >>= unmarshalNodes . maybeToList)
```

Comment:
One consequence of `unmarshalNodes` taking a `[Node]` argument is that a couple of instances no longer have to perform a `peekNode` (since the node in question is passed in).

What if `gunmarshalNode` took a `Node` argument? Would that enable us to 🔥 this `peekNode` call, and the similar one in one of the `GUnmarshalSum` instances?

Reply:
Good suggestion. 👍 Did this.